assets/src/edit-story/components/panels/link.js

```diff
@@ -82,29 +85,43 @@ function LinkPanel({ selectedElements, onSetProperties }) {
   );
   const handleSubmit = useCallback(
     (evt) => {
-      onSetProperties({ link: state?.url ? state : null });
+      onSetProperties({ link: state?.url ? { ...state } : null });
       if (evt) {
         evt.preventDefault();
       }
     },
     [state, onSetProperties]
   );
 
+  const {
+    actions: { getLinkMetadata },
+  } = useAPI();
+
   const canLink = selectedElements.length === 1 && !isFill;
-  const populateMetadata = useCallback(
-    debounce(300, async (/** url */) => {
-      // TODO(wassgha): Implement getting the page metadata
-    })
-  );
+
+  const [populateMetadata] = useDebouncedCallback((url) => {
+    if (!validUrl.isUri(url)) {
```

Comment:
What would this say about "www.example.com/"?

Reply:
The scheme is required, do we want to allow `www.example.com/` ? If so, can we assume https?

Reply:
yeah I'd say let's allow unprefixed, and assume https:// then.

Reply:
There are some nice utils for [urls](https://github.com/WordPress/gutenberg/tree/master/packages/url/src) in gutenberg - 
Might be useful here. Specially [prepend-http](https://github.com/WordPress/gutenberg/blob/master/packages/url/src/prepend-http.js)

Reply:
Already implemented my own but I'll keep an eye on that if we need something more complex! Thanks for the reference!